scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/vfs/Paths.scala

```diff
@@ -32,11 +30,6 @@ object Paths {
 
   def scalaToSemanticdb(path: RelativePath): RelativePath = {
     require(isScala(path))
-    RelativePath(
-      semanticDbPrefix +
-        path.toString
-          .stripPrefix(scalaPrefix)
-          .stripSuffix(scalaSuffix) +
-        semanticDbSuffix)
+    semanticDbPrefix.resolve(path.resolveSibling(_.stripSuffix(scalaSuffix) + semanticDbSuffix))
```

Comment:
I find the refactoring of logic in `Paths` hard to follow. Let's discuss a formulation that would work for both of us. What did you find unnice about the previous implementation?